components/workspace/hooks.tsx

```diff
@@ -25,15 +26,34 @@ export const useWorkspaceList = () => {
 
 export const useWorkspaceAddItemsMutation = () => {
   const queryClient = useQueryClient()
+  const toastId = useRef<number | string | null>(null)
   const mutation = useMutation<string[], unknown, string[], unknown>(
     async (items) => {
       return addToList(items)
     },
     {
-      onSuccess: (_, addedItems) => {
-        toast.success(
-          `${pluralize(addedItems.length, 'subject')} added to workspace. `,
-        )
+      onSuccess: (allItems, addedItems) => {
+        const message = `Attempted to add ${pluralize(
+          addedItems.length,
+          'subject',
+        )}, there are ${pluralize(
+          allItems.length,
+          'subject',
+        )} in your workspace now.`
+
+        if (!toastId.current) {
+          toastId.current = toast.success(message)
+        } else {
+          // Only when pass the onClose here because if we pass it during instantiation, the update essentially triggers onClose
+          // causing the toast to close indefinitely
```

Comment:
Doesn't updating twice also cause this behavior ?


Would keeping another reference help ?
```tsx
const toastNr = useRef<number>(0)
const toastId = useRef<number | string | null>(null)

// ...

onSuccess: () => {
  // ...

  const currentNr = ++toastNr.current
  const onClose = () => {
    // Only clear the toastId ref when the latest toast is closed
    if (toastNr.current === currentNr) toastId.current = null
  }

  // use onClose both for init and update
  // ...
},
```

Reply:
thanks ended up using the isActive check.